src/main/java/domain/LadderGame.java

```diff
@@ -0,0 +1,56 @@
+package domain;
+
+import domain.ladder.Ladder;
+import domain.ladder.Line;
+import domain.player.Player;
+import domain.player.Players;
+import domain.product.Product;
+import domain.product.Products;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class LadderGame {
+    public static final String PLAYER_COUNT_SAME_PRODUCT = "[ERROR] 상품의 개수와 플레이어의 수가 일치하지 않습니다.";
+    Products products;
+    Players players;
+    Ladder ladder;
+
+    public LadderGame(Players players, Products products, Ladder ladder) {
+        checkLadderGame(players, products);
+        this.players = players;
+        this.products = products;
+        this.ladder = ladder;
+    }
+
+    public Map<Player, Product> play() {
+        Map<Player, Product> result = new LinkedHashMap<>();
+        for (Player player : players.getPlayers()) {
+            result.put(player, products.productOfIndex(oneClimbTheLadder(player, ladder)));
+        }
+        return result;
+    }
+
+    private int oneClimbTheLadder(Player player, Ladder ladder) {
+        for (Line line : ladder.getLines()) {
+            climbTheLadderPlayerPosition(player, line);
+        }
+        return player.getPosition();
+    }
```

Comment:
```suggestion
    private int oneClimbTheLadder(Player player) {
        for (Line line : ladder.getLines()) {
            climbTheLadderPlayerPosition(player, line);
        }
        return player.getPosition();
    }
```

Ladder 를 별도로 넘겨줄 필요가 없을 것 같네요!

+ oneClimbTheLadder 라는 메소드 명을 개선해 보면 어떨까요? (one, the 와 같은 관사를 제거해보는 것은 어떨까요?)
https://kjwsx23.tistory.com/372

Reply:
생각해보니 넘겨줄 필요가 없었군요...